### PR TITLE
Add guidance on injecting/extracting propagation context for out of proc requests

### PR DESCRIPTION
**Pull request by dasiths**
As requested by @cijothomas this change explicitly sets the expectation of an instrumented library communicating over the wire with regardings to injecting and extracting the propagation context.

### CODE REVIEW

**Review by reyang — APPROVED** [review]
LGTM. Might need to fix the markdownlint warnings.

File: docs/trace/extending-the-sdk/README.md [hunk 1]
```diff
@@ -126,7 +126,8 @@ Writing an instrumentation library typically involves 3 steps.
...
-   in this case subscribes to the `EventSource` callbacks
+   in this case subscribes to the `EventSource` callbacks.
+     - [Context Propgation](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Api/README.md#context-propagation): If your library initiates out of process requests or accepts them, the library needs to [inject the `PropagationContext`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/6b7f2dd77cf9d37260a853fcc95f7b77e296065d/examples/MicroserviceExample/Utils/Messaging/MessageSender.cs) to outgoing requests and [extract the context](https://github.com/open-telemetry/opentelemetry-dotnet/blob/6b7f2dd77cf9d37260a853fcc95f7b77e296065d/examples/MicroserviceExample/Utils/Messaging/MessageReceiver.cs) and hydrate the Activity/Baggage upon receiving incoming requests. This is only required if you're using your own protocol to communicate over the wire. (i.e. If you're using an already instrumented HttpClient or GrpcClient, this is already provided to you and **do not require** injecting/extracting `PropagationContext` explicitly again.)
```
[the remark this answers is not in the document]
Comment (dasiths) [reply docs/trace/extending-the-sdk/README.md]: resolved

**Comment by dasiths** [comment]
> LGTM. Might need to fix the markdownlint warnings.

linting issues fixed as well.

**Review by cijothomas — APPROVED** [review]

**Comment by dasiths** [comment]
@cijothomas this can now be merged IMO. Thoughts?

**Pull request merged**